[BUG] Compilation failure when enabling UBL

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

Enabling UBL breaks compilation with the following errors:

```
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp: In static member function 'static void unified_bed_leveling::G29()':
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:308:10: warning: unused variable 'probe_deployed' [-Wunused-variable]
  308 |     bool probe_deployed = false;
      |          ^~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp: In static member function 'static void unified_bed_leveling::G29_handle_post_processing()':
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:432:9: error: 'probe_deployed' was not declared in this scope
  432 |     if (probe_deployed) {
      |         ^~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp: In static member function 'static bool unified_bed_leveling::G29_handle_phase_ops()':
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:309:35: error: 'probe_deployed' was not declared in this scope
  309 |     #define SET_PROBE_DEPLOYED(N) probe_deployed = N
      |                                   ^~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:475:9: note: in expansion of macro 'SET_PROBE_DEPLOYED'
  475 |         SET_PROBE_DEPLOYED(true);
      |         ^~~~~~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:309:35: error: 'probe_deployed' was not declared in this scope
  309 |     #define SET_PROBE_DEPLOYED(N) probe_deployed = N
      |                                   ^~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:513:11: note: in expansion of macro 'SET_PROBE_DEPLOYED'
  513 |           SET_PROBE_DEPLOYED(true);
      |           ^~~~~~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp: In static member function 'static void unified_bed_leveling::G29_handle_tilt_mesh()':
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:309:35: error: 'probe_deployed' was not declared in this scope
  309 |     #define SET_PROBE_DEPLOYED(N) probe_deployed = N
      |                                   ^~~~~~~~~~~~~~
Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp:709:5: note: in expansion of macro 'SET_PROBE_DEPLOYED'
  709 |     SET_PROBE_DEPLOYED(true);
      |     ^~~~~~~~~~~~~~~~~~
*** [.pio/build/GD32F103RET6_sovol_maple/src/src/feature/bedlevel/ubl/ubl_G29.cpp.o] Error 1
```

I think it is because the following code:

```cpp
  #ifdef EVENT_GCODE_AFTER_G29
    bool probe_deployed = false;
    #define SET_PROBE_DEPLOYED(N) probe_deployed = N
  #else
    #define SET_PROBE_DEPLOYED(N)
  #endif
```

is declared inside the `G29()` function, but is referenced in the if conditional of G29_handle_post_processing.

Note: I got a bunch of warnings, such as "#warning "This build does not have access to >= c++17 features."".

### Bug Timeline

_No response_

### Expected behavior

No compilation error should occur because of config change

### Actual behavior

_No response_

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

Sovol SV06

### Electronics

_No response_

### LCD/Controller

_No response_

### Other add-ons

_No response_

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [x] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[configs.gz](https://github.com/user-attachments/files/30609415/configs.gz)

--- a/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
+++ b/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
@@ -303,12 +303,12 @@
 G29_parameters_t unified_bed_leveling::param;
 
 #ifdef EVENT_GCODE_AFTER_G29
-  bool probe_deployed = false;
-  #define SET_PROBE_DEPLOYED(N) probe_deployed = N
+  bool ubl_probe_deployed = false;
+  #define SET_UBL_PROBE_DEPLOYED(N) ubl_probe_deployed = N
 #else
-  #define SET_PROBE_DEPLOYED(N)
+  #define SET_UBL_PROBE_DEPLOYED(N)
 #endif
-  
+
 void unified_bed_leveling::G29() {
   if (G29_parse_parameters()) return; // Abort on parameter error
 
@@ -428,7 +428,7 @@ void unified_bed_leveling::G29_handle_post_processing() {
 
   #ifdef EVENT_GCODE_AFTER_G29
     if (DEBUGGING(LEVELING)) DEBUG_ECHOLNPGM("After G29 G-code: ", EVENT_GCODE_AFTER_G29);
-    if (probe_deployed) {
+    if (ubl_probe_deployed) {
       planner.synchronize();
       gcode.process_subcommands_now(F(EVENT_GCODE_AFTER_G29));
     }
@@ -471,7 +471,7 @@ bool unified_bed_leveling::G29_handle_phase_ops() {
         probe_entire_mesh(param.XY_pos, parser.seen_test('T'), parser.seen_test('E'), parser.seen_test('U'));
 
         motion.report_position();
-        SET_PROBE_DEPLOYED(true);
+        SET_UBL_PROBE_DEPLOYED(true);
       } break;
 
     #endif // HAS_BED_PROBE
@@ -509,7 +509,7 @@ bool unified_bed_leveling::G29_handle_phase_ops() {
             SERIAL_ECHOLNPGM("?Error in Business Card measurement.");
             return false;
           }
-          SET_PROBE_DEPLOYED(true);
+          SET_UBL_PROBE_DEPLOYED(true);
         }
 
         if (!motion.can_reach(param.XY_pos)) {
@@ -694,6 +694,7 @@ void unified_bed_leveling::G29_handle_invalidate() {
 }
 
 #if HAS_BED_PROBE
+
   /**
    * Handle tilt mesh (J parameter)
    */
@@ -705,8 +706,9 @@ void unified_bed_leveling::G29_handle_invalidate() {
       motion.blocking_move_xy(0.5f * (mesh_min.x + mesh_max.x), 0.5f * (mesh_min.y + mesh_max.y));
     #endif
     motion.report_position();
-    SET_PROBE_DEPLOYED(true);
+    SET_UBL_PROBE_DEPLOYED(true);
   }
+
 #endif // HAS_BED_PROBE
 
 /**

--- a/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
+++ b/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
@@ -302,15 +302,14 @@
 
 G29_parameters_t unified_bed_leveling::param;
 
+#ifdef EVENT_GCODE_AFTER_G29
+  bool probe_deployed = false;
+  #define SET_PROBE_DEPLOYED(N) probe_deployed = N
+#else
+  #define SET_PROBE_DEPLOYED(N)
+#endif
+  
 void unified_bed_leveling::G29() {
-
-  #ifdef EVENT_GCODE_AFTER_G29
-    bool probe_deployed = false;
-    #define SET_PROBE_DEPLOYED(N) probe_deployed = N
-  #else
-    #define SET_PROBE_DEPLOYED(N)
-  #endif
-
   if (G29_parse_parameters()) return; // Abort on parameter error
 
   const uint8_t p_val = parser.byteval('P');

--- a/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
+++ b/Marlin/src/feature/bedlevel/ubl/ubl_G29.cpp
@@ -310,6 +310,8 @@ G29_parameters_t unified_bed_leveling::param;
 #endif
 
 void unified_bed_leveling::G29() {
+  SET_UBL_PROBE_DEPLOYED(false);
+
   if (G29_parse_parameters()) return; // Abort on parameter error
 
   const uint8_t p_val = parser.byteval('P');

--- a/Marlin/src/lcd/dwin/jyersui/dwin.cpp
+++ b/Marlin/src/lcd/dwin/jyersui/dwin.cpp
@@ -199,7 +199,7 @@ grid_count_t gridpoint;
 float corner_avg;
 float corner_pos;
 
-bool probe_deployed = false;
+bool dwin_probe_deployed = false;
 
 JyersDWIN jyersDWIN;
 
@@ -1268,8 +1268,8 @@ void JyersDWIN::menuItemHandler(const uint8_t menu, const uint8_t item, bool dra
             drawMenuItem(row, ICON_Back, GET_TEXT_F(MSG_BACK));
           else {
             #if HAS_BED_PROBE
-              probe_deployed = false;
-              probe.set_deployed(probe_deployed);
+              dwin_probe_deployed = false;
+              probe.set_deployed(dwin_probe_deployed);
             #endif
             drawMenu(ID_Prepare, PREPARE_MOVE);
           }
@@ -1329,12 +1329,12 @@ void JyersDWIN::menuItemHandler(const uint8_t menu, const uint8_t item, bool dra
           case MOVE_P:
             if (draw) {
               drawMenuItem(row, ICON_StockConfiguration, F("Probe"));
-              drawCheckbox(row, probe_deployed);
+              drawCheckbox(row, dwin_probe_deployed);
             }
             else {
-              FLIP(probe_deployed);
-              probe.set_deployed(probe_deployed);
-              drawCheckbox(row, probe_deployed);
+              FLIP(dwin_probe_deployed);
+              probe.set_deployed(dwin_probe_deployed);
+              drawCheckbox(row, dwin_probe_deployed);
             }
             break;
         #endif

